### PR TITLE
add python codec and converter docs

### PR DESCRIPTION
**Pull request by axfelix**
Add a Python SDK converter/codec section

### CODE REVIEW

**Review by flossypurse — COMMENTED** [review]
@rachfop for Python expertise.

File: docs-src/python/how-to-use-a-custom-payload-codec-in-python.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+**Create a custom Payload Codec**
+
+Custom Data Converters can change the default Temporal Data Conversion behavior by adding hooks, sending payloads to external storage, or performing different encoding steps.
+If you only need to change the encoding performed on your payloads -- by adding compression or encryption -- you can override the default Data Converter using by creating a new `PayloadCodec`.
+
+The `PayloadCodec` needs to implement `encode()` and `decode()` functions at a minimum.
+These should loop through all of a Workflow's payloads, perform all of your necessary marshalling, compression, or encryption steps in order, and set an `"encoding"` metadata field.
+Here is an example that marshals and then compresses a payload using Python's [cramjam](https://github.com/milesgranger/cramjam/tree/master/cramjam-python) library to provide `snappy` compression:
+
+```python
```
Comment (napcs) [inline docs-src/python/how-to-use-a-custom-payload-codec-in-python.md:20]: shouldn't this code be external to docs and imported in so we can test it?

**Review by napcs — CHANGES_REQUESTED** [review]

File: docs-src/python/how-to-use-a-custom-payload-codec-in-python.md
Comment (fairlydurable) [inline docs-src/python/how-to-use-a-custom-payload-codec-in-python.md:1]: marshalling -> marshaling 

Technically "Marshaling and marshalling are both English terms. Marshaling is predominantly used in 🇺🇸 American (US) English ( en-US ) while marshalling is predominantly used in 🇬🇧 British English (used in UK/AU/NZ) ( en-GB )."

File: docs-src/python/how-to-use-a-custom-payload-converter-in-python.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+Users are strongly encouraged to use a single `dataclass` for parameter and return types so fields with defaults can be easily added without breaking compatibility.
+Classes with generics may not have the generics properly resolved.
+The current implementation does not have generic type resolution. Users should use concrete types.
+
+### Custom Type Data Conversion
+
+When converting from JSON, Workflow and Activity type hints are taken into account to convert to the proper types.
+All common Python typings including `Optional`, `Union`, all forms of iterables and mappings, and `NewType` are supported in addition the regular JSON values mentioned before.
+
+In Pyhton, Data Converters contain a reference to a Payload Converter class that is used to convert input and output payloads.
```
Comment (fairlydurable) [inline docs-src/python/how-to-use-a-custom-payload-converter-in-python.md:45]:
```suggestion
In Python, Data Converters contain a reference to a Payload Converter class that is used to convert input and output payloads.
```

File: docs-src/python/how-to-use-a-custom-payload-codec-in-python.md (hunk 1 above)
Comment (axfelix) [reply docs-src/python/how-to-use-a-custom-payload-codec-in-python.md:20]: Yes... but I don't think we're using snipsync consistently enough in this repo right now, and most of these snippets are fairly abstracted and I don't think we could make them pass any CI from top to bottom.

**Comment by axfelix** [comment]
@rachfop If you have time to look at it, would be good to get this merged

**Review by flossypurse — APPROVED** [review]

**Pull request merged**